common/templates/options.php

```diff
@@ -146,11 +146,12 @@
                 <th scope="row"></th>
                 <td>
                     <label>
-                        GTIN meta key
+                        GTIN key
                         <select name="<?= $plugin->getOptionName('custom_gtin'); ?>">
-                            <option value=""><?= $plugin->hasActiveGtinPlugin() ? __('Automatic detection', 'webwinkelkeur') : 'Select key'; ?></option>
-                            <?php foreach ($plugin->getProductMetaKeys() as $key): ?>
-                                <option value="<?= $key; ?>" <?= $key == $config['custom_gtin'] ? 'selected' : ''; ?>><?= $key; ?></option>
+                            <option value=""><?= $plugin->getActiveGtinPlugin() ? __('Automatic detection', 'webwinkelkeur') . ' (' . (explode('/', $plugin->getActiveGtinPlugin())[0] ?? '') . ')' : 'Select key'; ?></option>
+                            <?php foreach ($plugin->getProductKeys() as $key): ?>
+                                <?php $option_value = $key['type'] . ':' . $key['name']; ?>
+                                <option value="<?= $option_value; ?>" <?= $option_value == $config['custom_gtin'] ? 'selected' : ''; ?>><?= $key['name']; ?></option>
```

Comment:
1. I would add `option_value` (= type + name) and `label` (= name + ( + type + )) on the `$key` result from `getProductKeys`, so you don't need that logic in the template. (Always better to keep templates dumb.)
1. You should apply `htmlentities` on anything that you didn't generate yourself and could possibly contain quotes, etc., so that is `$option_value` and `$key['name']` here.  (Even though of course meta names and attributes _shouldn't_ contain special characters, you can't guarantee it.)
1. Maybe it's good to add _(attribute)_ and _(meta key)_ after the name so that they can be distinguished.